src/gluonts/mx/model/estimator.py

```diff
@@ -35,6 +35,39 @@
 from gluonts.transform import Transformation, TransformedDataset
 
 
+class DataLoader(BaseModel):
+    num_workers: Optional[int]
+    num_prefetch: Optional[int]
+    shuffle_buffer_length: Optional[int]
+    cache_data: bool = False
+
+    def train(self, dataset, transform, batch_size, stack_fn, decode_fn):
+        if self.cache_data:
+            dataset = Cached(dataset)
+
+        return TrainDataLoader(
+            dataset=dataset,
+            transform=transform,
+            batch_size=batch_size,
+            stack_fn=stack_fn,
+            decode_fn=decode_fn,
+            num_workers=self.num_workers,
+            num_prefetch=self.num_prefetch,
+            shuffle_buffer_length=self.shuffle_buffer_length,
+        )
```

Comment:
Would it make sense to directly do here what happens in `TrainDataLoader`?

Also, this overloads the "data loader" terminology a bit, would it make sense to rename this class to something else, like `DataLoaderConfig` or something similar?

Reply:
For the first point: that would mean getting rid of these functions https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/dataset/loader.py#L226-L377 but we can also keep them around